src/osd/modules/render/drawbgfx.cpp

```diff
@@ -423,10 +457,11 @@ bool video_bgfx::set_platform_data(bgfx::PlatformData &platform_data, osd_window
 		platform_data.nwh = wmi.info.cocoa.window;
 		break;
 #endif
-#if defined(SDL_VIDEO_DRIVER_WAYLAND) && SDL_VERSION_ATLEAST(2, 0, 16)
+#if defined(SDL_VIDEO_DRIVER_WAYLAND) && SDL_VERSION_ATLEAST(2, 0, 16) && defined(SDLMAME_USE_WAYLAND)
 	case SDL_SYSWM_WAYLAND:
 		platform_data.ndt = wmi.info.wl.display;
-		platform_data.nwh = wmi.info.wl.egl_window;
+		platform_data.nwh = create_wl_egl_window(dynamic_cast<sdl_window_info const &>(window).platform_window(), wmi.info.wl.surface);
+		platform_data.type = bgfx::NativeWindowHandleType::Wayland;
```

Comment:
Because `create_wl_egl_window` can fail (unlike the other branches of this `if` statement), you need to check whether it succeeded here, and `return false` if it didn’t.

Reply:
Should be fixed now. Is it also needed for `sdlNativeWindowHandle()`?

Reply:
> Should be fixed now. Is it also needed for `sdlNativeWindowHandle()`?

It is, but the function would need to be rearranged a bit.  I’ll worry about that later.